sql/memo/join_order_builder.go

```diff
@@ -182,6 +180,103 @@ func (j *joinOrderBuilder) populateSubgraph(n sql.Node) (vertexSet, edgeSet, *Ex
 	return j.allVertices().difference(startV), j.allEdges().Difference(startE), group
 }
 
+// ensureClosure adds the closure of all transitive equivalency groups
+// to the join tree. Each transitive edge will add an inner edge, filter,
+// and join group that inherit join type and tree depth from the original
+// join tree.
+func (j *joinOrderBuilder) ensureClosure(grp *ExprGroup) {
+	fds := grp.RelProps.FuncDeps()
+	for _, set := range fds.Equiv().Sets() {
+		for col1, ok1 := set.Next(1); ok1; col1, ok1 = set.Next(col1 + 1) {
```

Comment:
Not really part of this PR, but `Next()` is usually part of an iterator pattern that doesn't usually take an arg

You probably want this to be called `Get`

Also use `hasNext` instead of `ok1` here